### PR TITLE
docs: cruise control overview edit

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
Updates [Cruise Control for cluster rebalancing](https://strimzi.io/docs/operators/in-development/configuring.html#cruise-control-concepts-str). 

- Adds a summary terms--goal types, proposals and modes-- to the intro
- Lists where you can specify goals
- Introduces the concepts of modes earlier in the optimization proposals introduction
- Updates headings for findability
- Changes discrete headings so they are linked

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/assemblies/cruise-control/assembly-cruise-control-concepts.adoc [hunk 1]
```diff
@@ -6,15 +6,49 @@
...
-You can deploy {CruiseControlProject} to your Strimzi cluster and use it to _rebalance_ the Kafka cluster.
-
 Cruise Control is an open source system for automating Kafka operations, such as monitoring cluster workload, rebalancing a cluster based on predefined constraints, and detecting and fixing anomalies.
 It consists of four main components--the Load Monitor, the Analyzer, the Anomaly Detector, and the Executor--and a REST API for client interactions.
-Strimzi utilizes the REST API to support the following Cruise Control features:
 
-* Generating _optimization proposals_ from multiple _optimization goals_.
+You can deploy {CruiseControlProject} to your Strimzi cluster and use it to _rebalance_ a Kafka cluster.
+You deploy Cruise Control through configuration of a `Kafka` resource.
+You perform rebalances through the `KafkaBalance` resource, which generates and applies optimization proposals.
```
Comment (ppatierno) [inline documentation/assemblies/cruise-control/assembly-cruise-control-concepts.adoc:14]: it's `KafkaRebalance` resource

File: documentation/assemblies/cruise-control/assembly-cruise-control-concepts.adoc [hunk 2]
```diff
@@ -6,15 +6,49 @@
...
+Optimization proposals::  Optimization proposals comprise the goals you want to achieve from a rebalance. 
+You generate an optimization proposal to create a summary of proposed changes and the results that are possible with the rebalance.
+The goals are assessed in a specific order of priority.
+You can then choose to approve or reject the proposal.
+You can reject the proposal to run it again with an adjusted set of goals.  
++
+You can generate an optimization proposal in one of three modes.
++
+--
+* *`full` mode* is the default mode and runs a full rebalance.
```
Comment (ppatierno) [inline documentation/assemblies/cruise-control/assembly-cruise-control-concepts.adoc:48]: do we need "mode" for each of them? Isn't it enough "full", "add-brokers" and "remove brokers" .. the you say "it is the mode". Anyway we are introducing them as "... three modes" before.

File: documentation/modules/cruise-control/con-optimization-proposals.adoc [hunk 3]
```diff
@@ -6,25 +6,59 @@
 
 = Optimization proposals overview
 
+[role="_abstract"]
+You use the `KafkaBalance` resource to generate and apply the changes suggested in optimization proposals.
```
Comment (ppatierno) [inline documentation/modules/cruise-control/con-optimization-proposals.adoc:10]: It's `KafkaRebalance` resource.

**Review by ppatierno — CHANGES_REQUESTED** [review]

File: documentation/assemblies/cruise-control/assembly-cruise-control-concepts.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/assemblies/cruise-control/assembly-cruise-control-concepts.adoc]: You're right. We don't need to highlight mode here. Removed.

**Review by ppatierno — APPROVED** [review]

**Review by tomncooper — APPROVED** [review]
LGTM

**Pull request merged**